pyresample/geometry.py

```diff
@@ -1423,10 +1901,119 @@ def copy(self, **override_kwargs):
         width = int(self.width / dims.get('x', 1))
         height = int(self.height / dims.get('y', 1))
         return self.copy(height=height, width=width)
 
+    def downsample(self, x=1, y=1):
+        """Return a downsampled version of the area."""
+        # Check input validity
+        x = int(x)
+        y = int(y)
+        if x == 1 and y == 1:
+            return self
+        if x < 1 or y < 1:
+            raise ValueError('x and y arguments must be positive integers larger or equal to 1.')
+        # Downsample
+        width = int(self.width / x)
+        height = int(self.height / y)
+        return self.copy(height=height, width=width)
+
+    def upsample(self, x=1, y=1):
+        """Return an upsampled version of the area."""
+        # Check input validity
+        x = int(x)
+        y = int(y)
+        if x == 1 and y == 1:
+            return self
+        if x < 1 or y < 1:
+            raise ValueError('x and y arguments must be positive integers larger or equal to 1.')
+        # Upsample
+        width = int(self.width * x)
+        height = int(self.height * y)
+        return self.copy(height=height, width=width)
+
+    def extend(self, x=0, y=0):
+        """Extend AreaDef by x/y pixels on each side."""
+        if self.is_geostationary:
+            raise NotImplementedError("'extend' method is not implemented for GEO AreaDefinition.")
+        # Check input validity
+        x = int(x)
+        y = int(y)
+        if x < 0 or y < 0:
+            raise ValueError('x and y arguments must be positive integers.')
+        if x == 0 and y == 0:
+            return self
+        # Retrieve pixel and area info
+        new_width = self.width + 2 * x
+        new_height = self.height + 2 * y
+        pixel_size_x = self.pixel_size_x
+        pixel_size_y = self.pixel_size_y
+        # Extend area_extent (lower_left_x, lower_left_y, upper_right_x, upper_right_y)
+        area_extent = self._area_extent
+        new_area_extent = list(area_extent)
+        new_area_extent[0] = new_area_extent[0] - pixel_size_x * x
+        new_area_extent[1] = new_area_extent[1] - pixel_size_y * y
+        new_area_extent[2] = new_area_extent[2] + pixel_size_x * x
+        new_area_extent[3] = new_area_extent[3] + pixel_size_y * y
+        # Define new AreaDefinition
+        projection = self.crs_wkt
+        area_def = AreaDefinition(self.area_id, self.description, self.proj_id,
+                                  projection=projection,
+                                  width=new_width,
+                                  height=new_height,
+                                  area_extent=new_area_extent,
+                                  rotation=self.rotation,
+                                  nprocs=self.nprocs,
+                                  dtype=self.dtype)
+
+        return area_def
+
+    def reduce(self, x=0, y=0):
```

Comment:
I think this is already implemented with the more pythonic `new_area = area[slice(-2, 2), slice(10, -4)]`